### ISSUE TITLE
Implement weight tying in morpheme-aware training model

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#51]
## Problem

The current `MorphemeReconstructionModel` in `scripts/train_morpheme_aware.py` has ~1.29M parameters because it uses separate linear layers for reconstruction heads:

- Embeddings: ~640K params (5K roots × 128d)
- Reconstruction heads: ~653K params (separate classifiers)
- **Total: ~1.29M params**

This doubles the parameter count unnecessarily.

## Solution

Implement **weight tying** between the embedding layers and reconstruction heads. This is a standard technique in masked language modeling (used in BERT, GPT, etc.) where:

1. The root embedding matrix is transposed and used as the classifier weights
2. Only bias parameters are separate
3. Gradients flow through shared weights during backprop

## Benefits

✅ **50% fewer parameters** (~640K instead of ~1.29M)
✅ **10-20% faster training** (fewer params to update)
✅ **Better generalization** (weight sharing acts as regularization)
✅ **Lower memory usage** (smaller optimizer state)
✅ **Theoretically sound** (embedding and classifier should represent same concepts)

## Implementation

Modify `MorphemeReconstructionModel.__init__()` (scripts/train_morpheme_aware.py:321-334):

**Current:**
```python
self.root_classifier = nn.Linear(hidden_dim, len(compositional_embedding.root_vocab))
self.prefix_classifier = nn.Linear(hidden_dim, len(compositional_embedding.prefix_vocab))
self.suffix_classifier = nn.Linear(hidden_dim, len(compositional_embedding.suffix_vocab))
```

**With weight tying:**
```python
# Tie weights: reuse embedding matrices as classifier weights
# Only need separate bias parameters
self.root_classifier_bias = nn.Parameter(torch.zeros(len(compositional_embedding.root_vocab)))
self.prefix_classifier_bias = nn.Parameter(torch.zeros(len(compositional_embedding.prefix_vocab)))
self.suffix_classifier_bias = nn.Parameter(torch.zeros(len(compositional_embedding.suffix_vocab)))
```

**In forward() method:**
```python
# Use transposed embeddings as classifier weights
root_logits = F.linear(context, self.embedding.root_embeddings.weight, self.root_classifier_bias)
prefix_logits = F.linear(context, self.embedding.prefix_embeddings.weight, self.prefix_classifier_bias)
suffix_logits = F.linear(context, self.embedding.suffix_embeddings.weight, self.suffix_classifier_bias)
```

## Testing

After implementation:
1. Run with `--test` flag and verify parameter count is ~640K (not ~1.29M)
2. Train for a few epochs and verify loss decreases normally
3. Compare training speed before/after

## References

- Standard practice in BERT, GPT, and most masked language models
- See `train_semantic_similarity.py` for similar pattern
- Research: "Using the Output Embedding to Improve Language Models" (Press & Wolf, 2017)

## Priority

**HIGH** - This should be done before full training runs to save compute time and improve model quality.

**Comment by marctjones** [comment]
Superseded by Issue #57 - the entire MLM approach is wrong
